api/src/test/java/org/openmrs/module/bedmanagement/BedManagementDaoImplTest.java

```diff
@@ -0,0 +1,178 @@
+package org.openmrs.module.bedmanagement;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.openmrs.Encounter;
+import org.openmrs.EncounterType;
+import org.openmrs.Patient;
+import org.openmrs.Visit;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.bedmanagement.dao.BedManagementDao;
+import org.openmrs.module.bedmanagement.entity.Bed;
+import org.openmrs.module.bedmanagement.entity.BedPatientAssignment;
+import org.openmrs.module.bedmanagement.entity.BedType;
+import org.openmrs.test.BaseModuleContextSensitiveTest;
+
+import java.util.Date;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+public class BedManagementDaoImplTest extends BaseModuleContextSensitiveTest {
+	
+	private BedManagementDao bedManagementDao;
+	
+	private EncounterType defaultEncounterType;
+	
+	@Before
+	public void setup() throws Exception {
+		executeDataSet("org/openmrs/include/standardTestDataset.xml");
+		
+		bedManagementDao = Context.getRegisteredComponent("bedManagementDao", BedManagementDao.class);
+		
+		defaultEncounterType = Context.getEncounterService().getEncounterType("Scheduled");
+	}
+	
+	private Bed createBed(String bedNumber) {
+		BedType bedType = new BedType();
+		bedType.setName("Test BedType");
+		bedType.setDisplayName("TB");
+		bedType.setDescription("Test Bed Type");
+		bedType = bedManagementDao.saveBedType(bedType);
+		
+		Bed bed = new Bed();
+		bed.setBedNumber(bedNumber);
+		bed.setStatus("AVAILABLE");
+		bed.setBedType(bedType);
+		
+		return bedManagementDao.saveBed(bed);
+	}
+	
+	private Visit getTestVisit(Patient patient, Integer visitId) {
+		Visit visit = Context.getVisitService().getVisit(visitId);
+		assertNotNull(visit);
+		return visit;
+	}
+	
+	private Encounter getTestEncounter(Patient patient, Integer encounterId) {
+		Encounter encounter = Context.getEncounterService().getEncounter(encounterId);
+		assertNotNull(encounter);
+		return encounter;
+	}
+	
+	@Test
+	public void shouldReturnBedByPatient() {
+		Patient patient = Context.getPatientService().getPatient(2);
+		Bed bed = createBed("B101");
+		
+		Encounter encounter = getTestEncounter(patient, 3);
+		
+		BedPatientAssignment assignment = new BedPatientAssignment();
+		assignment.setBed(bed);
+		assignment.setPatient(patient);
```

Comment:
Instead of doing all this, just have this data in the test dataset file and then it gets loaded as expected.